pkg/services/star/starimpl/star.go

```diff
@@ -0,0 +1,39 @@
+package starimpl
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/services/star"
+)
+
+type serviceImpl struct {
+	starStore store
+}
+
+func ProvideService(sqlstore sqlstore.Store) star.Service {
+	m := &serviceImpl{starStore: newStarStore(sqlstore)}
+	return m
+}
+
+func (m *serviceImpl) StarDashboard(ctx context.Context, cmd *star.StarDashboardCommand) error {
+	if cmd.DashboardId == 0 || cmd.UserId == 0 {
+		return star.ErrCommandValidationFailed
+	}
```

Comment:
I'm open to a discussion here, but what if we move such validations into StarDashboardCommand.Validate() method? And would be calling the method here instead? Or maybe we should validate at the API layer? I don't have a strong opinion here, but it's interesting where validation should better belong.

Reply:
done with the Validate as method of cmd